homeassistant/components/sonos/__init__.py

```diff
@@ -338,19 +339,12 @@ async def async_poll_manual_hosts(
         self, now: datetime.datetime | None = None
     ) -> None:
         """Add and maintain Sonos devices from a manual configuration."""
-
-        def get_sync_attributes(soco: SoCo) -> set[SoCo]:
-            """Ensure I/O attributes are cached and return visible zones."""
-            _ = soco.household_id
-            _ = soco.uid
-            return soco.visible_zones
-
         for host in self.hosts:
-            ip_addr = socket.gethostbyname(host)
+            ip_addr = await self.hass.async_add_executor_job(socket.gethostbyname, host)
```

Comment:
It looks like these could already be ip addresses. We could use `from homeassistant.util.network import is_ip_address` here and avoid the executor job in that case as the gethostbyname would be a no-op in that case and the cost of the executor job would vastly exceed doing the no-op

Reply:
There are plenty occasions of gethostbyname in other integrations that are already wrapped in an executor. Same would apply for those. 

Should we add a helper function in `homeassistant.util.network`like `resolve_ip_4` doing this for all integrations?

Reply:
That sounds like a good idea. We already have a problem with executor getting overloaded for some users so anything we can do to mitigate it would be a win

Reply:
You might as well cache the lookups while you're at it (although your router is probably doing that already).

Probably not local subnet lookups, though, since those can changed through DHCP/mDNS

Reply:
I would advise against adding caching in a patch version since cache expiration is never simple.  

Aiohttp already has a dns caching built in. I would try to reuse that if it's possible, but again let's no go down this path now as it's more important to solve the blocking IO in the event loop in a patch

Reply:
Should we then merge this as is and address the helper in a new PR?

Reply:
That makes sense to me.